### PR TITLE
First basic implementation

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Allows to import a PHP data set.
It will be transformed to an expected csv file and stored in temp folder.
It than can be passed to existing API.

This is a first step, not the expected final result.

### CODE REVIEW

File: Classes/TestingFramework.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                    }
+                    if (isset($assertion['uid'])) {
+                        $recordIdentifier = $tableName . ':' . $assertion['uid'];
+                        $additionalInformation = $this->renderRecords($assertion, $records[$assertion['uid']]);
+                    } else {
+                        $recordIdentifier = $tableName;
+                        $additionalInformation = $this->arrayToString($assertion);
+                    }
+
+                    $failMessages[] = 'Assertion in data-set failed for "' . $recordIdentifier . '":' . PHP_EOL . $additionalInformation;
```
Comment (justusmoroni) [inline Classes/TestingFramework.php:75]: We use `data-set` here and `data set` in line 42. Let's use the same term and wording in both cases?

File: Classes/TestingFramework.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                // Unset asserted record
+                unset($records[$result]);
+                // Increase assertion counter
+                self::assertTrue($result !== false);
+            }
+
+            if (!empty($records)) {
+                foreach ($records as $record) {
+                    $recordIdentifier = $tableName . ':' . $record['uid'];
+                    $failMessages[] = 'Not asserted record found for "' . $recordIdentifier . '".';
```
Comment (justusmoroni) [inline Classes/TestingFramework.php:88]: Sounds strange

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+   .. code-block:: php
+
+      $this->assertPHPDataSet(__DIR__ . '/Fixtures/SimpleSet.php');
+
+TODO
+====
+
+#. Implement use case to check for necessary updates and allow updates.
+   Use full for static data during deployment within update wizards or other scripts.
```
Comment (justusmoroni) [inline README.rst]: The beginning seems wrong. What is meant here?

File: Classes/TestingFramework.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+        if (!empty($failMessages)) {
+            self::fail(implode(PHP_EOL, $failMessages));
+        }
+    }
+
+    private function ensureFileExists(string $filePath): void
+    {
+        if (file_exists($filePath) === false) {
+            throw new InvalidArgumentException('The requestes PHP data set file "' . $filePath . '" does not exist.', 1681207108);
```
Comment (justusmoroni) [inline Classes/TestingFramework.php]:
```suggestion
            throw new InvalidArgumentException('The requested PHP data set file "' . $filePath . '" does not exist.', 1681207108);
```

File: phpunit.xml.dist [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    convertErrorsToExceptions="true"
+    convertWarningsToExceptions="true"
+    forceCoversAnnotation="false"
+    processIsolation="false"
+    stopOnError="false"
+    stopOnFailure="false"
+    stopOnIncomplete="false"
+    stopOnSkipped="false"
+    verbose="false"
+    >
```
Comment (justusmoroni) [inline phpunit.xml.dist]: Maybe?

```suggestion
>
```

File: shell.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      php
+      composer
+    ];
+    text = ''
+      composer install --prefer-dist --no-progress --working-dir="$PROJECT_ROOT"
+    '';
+  };
+
+in pkgs.mkShell {
+  name = "TYPO3 Package testing framework php datasets";
```
Comment (justusmoroni) [inline shell.nix]: Isn't this the old repository name?

**Review by justusmoroni — CHANGES_REQUESTED** [review]

File: Classes/TestingFramework.php (hunk 1 above)
Comment (DanielSiepmann) [reply Classes/TestingFramework.php:75]: Will use `data-set` to be in line with testing framework.

File: Classes/TestingFramework.php (hunk 2 above)
Comment (DanielSiepmann) [reply Classes/TestingFramework.php:88]: That's copied from testing framework. What's strange? There is a not asserted record, sounds fine to me.

File: README.rst (hunk 3 above)
Comment (DanielSiepmann) [reply README.rst]: `full` was a left over.

File: Classes/TestingFramework.php (hunk 2 above)
Comment (justusmoroni) [reply Classes/TestingFramework.php:88]: Ok - let's keep it like it is.

**Review by justusmoroni — APPROVED** [review]

**Pull request merged**